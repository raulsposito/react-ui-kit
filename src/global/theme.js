// https://www.color-hex.com/color/8293ee

export const colors = {
  primary: {
    regular: '#FC760D',
    dark: '#E46B0B',
    darker: '#C66F2B',
    hover: '#E17622',
    focus: '#B97641',
    selected: '#D4762C',
    disabled: '#F99546'
  },
  variant: {
    regular: '#FCB10D',
    dark: '#E29F0C',
    darker: '#C9A557',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  secondary: {
    regular: '#2FCFCF',
    dark: '#1F9A9A',
    darker: '#147676',
    hover: '#3DBDBD',
    focus: '#52A2A2',
    selected: '#44B4B4',
    disabled: '#5FD8D8'
  },
  success: {
    regular: '#3DB265',
    dark: '#48A668',
    darker: '#59946D',
    hover: '#48A668',
    focus: '#59946D',
    selected: '#59946D',
    disabled: '#9EBCA8'
  },
  error: {
    regular: '#FF5668',
    dark: '#E45C6B',
    darker: '#BA666F',
    hover: '#E45C6B',
    focus: '#BA666F',
    selected: '#BA666F',
    disabled: '#F8A4AC'
  },
  warning: {
    regular: '#FFAF00',
    dark: '#E5BB4D',
    darker: '#CCA644',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  contrast: {
    regular: '#3C3C3C',
    disabled: '#E4E4E4'
  },
  system: {
    regular: '#BAB5B4',
    dark: '#CBC6C4',
    darker: '#D7D4D1',
    hover: '#ACA8A8',
    focus: '#989595',
    selected: '#A6A2A1',
    disabled: '#C8C4C3'
  },
  background: {
    primary: '#FFFFFF',
    regular: '#FFFFFF',
    hover: '#F1F1F1',
    disabled: '#F8F8F8'
  },
  events: {
    inprogress: '#3DB265',
    next: '#5678CF',
    finished: '#BAB5B4',
    suspended: '#FFD056',
    cancelled: '#FF5668',
    white: '#FFFFFF'
  }
}

export const spacing = {
  quarter: '2px',
  half: '4px',
  threeQuarters: '6px',
  one: '8px',
  oneAndAQuarter: '10px',
  oneAndAHalf: '12px',
  two: '16px',
  twoAndAQuarter: '18px',
  twoAndAHalf: '20px',
  three: '24px',
  threeAndAHalf: '28px',
  four: '32px',
  fourAndAHalf: '36px',
  five: '40px',
  fiveAndAHalf: '44px',
  six: '48px',
  sixAndAHalf: '52px',
  seven: '56px',
  sevenAndAHalf: '60px',
  eight: '64px',
  eightAndAHalf: '68px',
  nine: '72px',
  nineAndAHalf: '76px',
  ten: '80px',
  tenAndAHalf: '84px',
  eleven: '88px',
  elevenAndAHalf: '92px',
  twelve: '96px',
  twelveAndAHalf: '100px',
  thirteen: '104px',
  fourteen: '112x',
  fifteen: '120px'
}

export const page = {
  width: '1200px'
}
