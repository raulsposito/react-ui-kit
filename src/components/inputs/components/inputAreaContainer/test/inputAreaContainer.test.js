import React from 'react'
import { act, fireEvent, render, screen } from '@testing-library/react'

import { colors } from '@global/theme'

import InputAreaContainer from '..'

const mockData = {
  name: 'inputAreaContainer',
  placeholder: 'textarea placeholder',
  value: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
  error: 'Error Text',
  maxLength: 200,
  rows: 50,
  onChange: jest.fn(),
  onFocus: jest.fn(),
  onBlur: jest.fn()
}

describe('<InputAreaContainer />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <InputAreaContainer
        name={mockData.name}
        placeholder={mockData.placeholder}
        value={mockData.value}
        maxLength={mockData.maxLenght}
        rows={mockData.rows}
        onChange={mockData.onChange}
        onFocus={mockData.onFocus}
        onBlur={mockData.onBlur}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  describe('Disabled', () => {
    it('is disabled', () => {
      render(
        <InputAreaContainer
          name={mockData.name}
          placeholder={mockData.placeholder}
          value={mockData.value}
          maxLength={mockData.maxLenght}
          rows={mockData.rows}
          onChange={mockData.onChange}
          onFocus={mockData.onFocus}
          onBlur={mockData.onBlur}
          disabled
        />
      )

      const input = screen.queryByPlaceholderText(mockData.placeholder)
      expect(input).toBeDisabled()
    })
  })

  describe('Required', () => {
    it('is required', () => {
      render(
        <InputAreaContainer
          name={mockData.name}
          placeholder={mockData.placeholder}
          value={mockData.value}
          maxLength={mockData.maxLenght}
          rows={mockData.rows}
          onChange={mockData.onChange}
          onFocus={mockData.onFocus}
          onBlur={mockData.onBlur}
          required
        />
      )

      const input = screen.queryByPlaceholderText(mockData.placeholder)
      expect(input).toBeRequired()
    })
  })

  describe('Placeholder', () => {
    it('input has placeholder', () => {
      render(
        <InputAreaContainer
          name={mockData.name}
          placeholder={mockData.placeholder}
          value={mockData.value}
          maxLength={mockData.maxLenght}
          rows={mockData.rows}
          onChange={mockData.onChange}
          onFocus={mockData.onFocus}
          onBlur={mockData.onBlur}
        />
      )

      const input = screen.queryByPlaceholderText(mockData.placeholder)
      expect(input).toBeInTheDocument()
    })
  })

  describe('Name', () => {
    it('input has name', () => {
      render(
        <InputAreaContainer
          name={mockData.name}
          placeholder={mockData.placeholder}
          value={mockData.value}
          maxLength={mockData.maxLenght}
          rows={mockData.rows}
          onChange={mockData.onChange}
          onFocus={mockData.onFocus}
          onBlur={mockData.onBlur}
        />
      )

      const input = screen.queryByPlaceholderText(mockData.placeholder, { name: mockData.name })
      expect(input).toBeInTheDocument()
    })
  })

  describe('Input event', () => {
    it('calls function on change', () => {
      render(
        <InputAreaContainer
          name={mockData.name}
          placeholder={mockData.placeholder}
          value={mockData.value}
          maxLength={mockData.maxLenght}
          rows={mockData.rows}
          onChange={mockData.onChange}
          onFocus={mockData.onFocus}
          onBlur={mockData.onBlur}
        />
      )

      const input = screen.queryByPlaceholderText(mockData.placeholder)
      fireEvent.change(input, { target: { value: 'new value' } })
      expect(mockData.onChange).toHaveBeenCalledTimes(1)
    })

    it('calls function on focus', () => {
      render(
        <InputAreaContainer
          name={mockData.name}
          placeholder={mockData.placeholder}
          value={mockData.value}
          maxLength={mockData.maxLenght}
          rows={mockData.rows}
          onChange={mockData.onChange}
          onFocus={mockData.onFocus}
          onBlur={mockData.onBlur}
        />
      )

      const input = screen.queryByPlaceholderText(mockData.placeholder)
      fireEvent.focus(input)
      expect(mockData.onFocus).toHaveBeenCalledTimes(1)
    })

    it('calls function on blur', () => {
      render(
        <InputAreaContainer
          name={mockData.name}
          placeholder={mockData.placeholder}
          value={mockData.value}
          maxLength={mockData.maxLenght}
          rows={mockData.rows}
          onChange={mockData.onChange}
          onFocus={mockData.onFocus}
          onBlur={mockData.onBlur}
        />
      )

      const input = screen.queryByPlaceholderText(mockData.placeholder)
      fireEvent.blur(input)
      expect(mockData.onBlur).toHaveBeenCalledTimes(1)
    })
  })

  describe('Handle State Color', () => {
    describe('Handle one state at time', () => {
      it('has not state changes', () => {
        render(
          <InputAreaContainer
            name={mockData.name}
            placeholder={mockData.placeholder}
            value={mockData.value}
            maxLength={mockData.maxLenght}
            rows={mockData.rows}
            onChange={mockData.onChange}
            onFocus={mockData.onFocus}
            onBlur={mockData.onBlur}
          />
        )

        const inputContainer = screen.queryByPlaceholderText(mockData.placeholder)
        expect(inputContainer).toHaveStyle({ borderColor: `${colors.system.regular}` })
        expect(inputContainer).toHaveStyle({ backgroundColor: `${colors.background.regular}` })
      })

      it('is disabled', () => {
        render(
          <InputAreaContainer
            name={mockData.name}
            placeholder={mockData.placeholder}
            value={mockData.value}
            maxLength={mockData.maxLenght}
            rows={mockData.rows}
            onChange={mockData.onChange}
            onFocus={mockData.onFocus}
            onBlur={mockData.onBlur}
            disabled
          />
        )

        const inputContainer = screen.queryByPlaceholderText(mockData.placeholder)
        expect(inputContainer).toHaveStyle({ borderColor: `${colors.system.disabled}` })
        expect(inputContainer).toHaveStyle({ backgroundColor: `${colors.background.disabled}` })
      })

      it('has error', () => {
        render(
          <InputAreaContainer
            name={mockData.name}
            placeholder={mockData.placeholder}
            value={mockData.value}
            maxLength={mockData.maxLenght}
            rows={mockData.rows}
            onChange={mockData.onChange}
            onFocus={mockData.onFocus}
            onBlur={mockData.onBlur}
            error={mockData.error}
          />
        )

        const inputContainer = screen.queryByPlaceholderText(mockData.placeholder)
        expect(inputContainer).toHaveStyle({ borderColor: `${colors.error.regular}` })
        expect(inputContainer).toHaveStyle({ backgroundColor: `${colors.background.regular}` })
      })

      it('is focused', () => {
        render(
          <InputAreaContainer
            name={mockData.name}
            placeholder={mockData.placeholder}
            value={mockData.value}
            maxLength={mockData.maxLenght}
            rows={mockData.rows}
            onChange={mockData.onChange}
            onFocus={mockData.onFocus}
            onBlur={mockData.onBlur}
          />
        )

        const input = screen.queryByPlaceholderText(mockData.placeholder)
        act(() => input.focus())

        expect(input).toHaveFocus()
        // :focus:within pseudoclass used to color the border of external div of input can't be tested
        // expect(inputContainer).toHaveStyle({ borderColor: `${colors.primary.regular}` })
        expect(input).toHaveStyle({ backgroundColor: `${colors.background.regular}` })
      })
    })

    describe('Handle multiple states at time', () => {
      it('is disabled and has error', () => {
        render(
          <InputAreaContainer
            name={mockData.name}
            placeholder={mockData.placeholder}
            value={mockData.value}
            maxLength={mockData.maxLenght}
            rows={mockData.rows}
            onChange={mockData.onChange}
            onFocus={mockData.onFocus}
            onBlur={mockData.onBlur}
            error={mockData.error}
            disabled
          />
        )

        const inputContainer = screen.queryByPlaceholderText(mockData.placeholder)
        expect(inputContainer).toHaveStyle({ borderColor: `${colors.error.regular}` })
        expect(inputContainer).toHaveStyle({ backgroundColor: `${colors.background.disabled}` })
      })

      it('is disabled and focused', () => {
        render(
          <InputAreaContainer
            name={mockData.name}
            placeholder={mockData.placeholder}
            value={mockData.value}
            maxLength={mockData.maxLenght}
            rows={mockData.rows}
            onChange={mockData.onChange}
            onFocus={mockData.onFocus}
            onBlur={mockData.onBlur}
            disabled
          />
        )

        const input = screen.queryByPlaceholderText(mockData.placeholder)
        act(() => input.focus())

        expect(input).toHaveStyle({ borderColor: `${colors.system.disabled}` })
        expect(input).toHaveStyle({ backgroundColor: `${colors.background.disabled}` })
      })

      it('is disabled, focused and has error', () => {
        render(
          <InputAreaContainer
            name={mockData.name}
            placeholder={mockData.placeholder}
            value={mockData.value}
            maxLength={mockData.maxLenght}
            rows={mockData.rows}
            onChange={mockData.onChange}
            onFocus={mockData.onFocus}
            onBlur={mockData.onBlur}
            error={mockData.error}
            disabled
          />
        )

        const input = screen.queryByPlaceholderText(mockData.placeholder)
        act(() => input.focus())

        expect(input).toHaveStyle({ borderColor: `${colors.error.regular}` })
        expect(input).toHaveStyle({ backgroundColor: `${colors.background.disabled}` })
      })
    })
  })

  describe('TextArea Properties', () => {
    it('max length', () => {
      render(
        <InputAreaContainer
          name={mockData.name}
          placeholder={mockData.placeholder}
          value={mockData.value}
          maxLength={mockData.maxLength}
          rows={mockData.rows}
          onChange={mockData.onChange}
          onFocus={mockData.onFocus}
          onBlur={mockData.onBlur}
        />
      )

      const input = screen.queryByPlaceholderText(mockData.placeholder)
      expect(input).toHaveAttribute('maxLength', (mockData.maxLength).toString())
    })

    it('rows', () => {
      render(
        <InputAreaContainer
          name={mockData.name}
          placeholder={mockData.placeholder}
          value={mockData.value}
          maxLength={mockData.maxLength}
          rows={mockData.rows}
          onChange={mockData.onChange}
          onFocus={mockData.onFocus}
          onBlur={mockData.onBlur}
        />
      )

      const input = screen.queryByPlaceholderText(mockData.placeholder)
      expect(input).toHaveAttribute('rows', (mockData.rows).toString())
    })

    it('resize', () => {
      render(
        <InputAreaContainer
          name={mockData.name}
          placeholder={mockData.placeholder}
          value={mockData.value}
          maxLength={mockData.maxLenght}
          rows={mockData.rows}
          onChange={mockData.onChange}
          onFocus={mockData.onFocus}
          onBlur={mockData.onBlur}
          resize
        />
      )

      const input = screen.queryByPlaceholderText(mockData.placeholder)
      expect(input).toHaveStyle({ resize: 'vertical' })
    })
  })
})
