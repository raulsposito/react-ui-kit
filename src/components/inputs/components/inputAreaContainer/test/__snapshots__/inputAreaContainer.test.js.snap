// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputAreaContainer /> should match snapshot 1`] = `
<DocumentFragment>
  <textarea
    class="sc-bdfBQB HkEli"
    name="inputAreaContainer"
    placeholder="textarea placeholder"
    rows="50"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit
  </textarea>
</DocumentFragment>
`;
